Vector.cpp

```diff
@@ -0,0 +1,238 @@
+#include <cassert>
+#include <iostream>
+
+class Custom_Vector
+{
+private:
+    size_t _size{};
+    size_t _capacity{};
+    int *buffer;
+
+public:
+    Custom_Vector()
+        : _size(0),
+          _capacity(0),
+          buffer(nullptr) {}
+
+    explicit Custom_Vector(size_t size)
+        : _size(size),
+          _capacity(size),
+          buffer(new int[_capacity])
+    {
+        std::fill_n(buffer, _size, int());
+    }
+
+    Custom_Vector(const std::initializer_list<int> list)
+        : Custom_Vector(list.size())
+    {
+        std::copy(list.begin(), list.end(), buffer);
+    }
+
+    Custom_Vector(const Custom_Vector &other)
+        : Custom_Vector(other._size)
+    {
+        std::copy(other.buffer, other.buffer + other._size, buffer);
+    }
+
+    Custom_Vector(Custom_Vector &&other)
+        : Custom_Vector()
+    {
+        std::swap(buffer, other.buffer);
+        std::swap(_size, other._size);
+        std::swap(_capacity, other._capacity);
+    }
+
+    ~Custom_Vector() { delete[] buffer; }
+
+    size_t size() const
+    {
+        return _size;
+    }
+
+    size_t capacity() const
+    {
+        return _capacity;
+    }
+
+    void assign(size_t count, const int &value)
+    {
+        clear();
+        reserve(count);
+
+        _size = count;
+
+        std::fill_n(buffer, _size, value);
+    }
+
+    void assign(std::initializer_list<int> list)
+    {
+        clear();
+
+        reserve(list.size());
+
+        _size = list.size();
+
+        std::copy(list.begin(), list.end(), buffer);
+    }
+
+    void clear()
+    {
+        _size = 0;
+    }
+
+    void push_back(const int &element)
+    {
+        if (_capacity == 0)
+        {
+            reserve(1);
+        }
+        if (_size == _capacity)
+        {
+            reserve(2 * _capacity);
+        }
+
+        buffer[_size] = element;
+        _size++;
+    }
+
+    void reserve(size_t size_n)
+    {
+        if (size_n <= _capacity)
+        {
+            return;
+        }
+
+        int *point = new int[size_n];
+
+        std::copy(buffer, buffer + _size, point);
+
+        delete[] buffer;
+        std::swap(buffer, point);
+        _capacity = size_n;
+    }
+
+    int at(size_t pos)
+    {
+        if (_size - 1 < pos || _size == 0)
```

Comment:
When `_size - 1` is executed and the vector is empty the behavior is undefined. You should check emptiness before. Actually, cppreference notes the correct condition when the exception must be thrown. 

You also have this condition in other places. Check all of them, please.